internal/function_time_parser.go

```diff
@@ -846,26 +891,17 @@ func quarterFormatter(t *time.Time) ([]rune, error) {
 }
 
 func hourMinuteParser(text []rune, t *time.Time) (int, error) {
-	fmtLen := len("00:00")
-	if len(text) < fmtLen {
-		return 0, fmt.Errorf("unexpected hour:minute format")
-	}
-	splitted := strings.Split(string(text[:fmtLen]), ":")
-	if len(splitted) != 2 {
-		return 0, fmt.Errorf("unexpected hour:minute format")
-	}
-	hour := splitted[0]
-	minute := splitted[1]
-	if len(hour) != 2 || len(minute) != 2 {
-		return 0, fmt.Errorf("unexpected hour:minute format")
-	}
-	h, err := strconv.ParseInt(hour, 10, 64)
+	hProgress, h, err := parseDigitRespectingOptionalPlaces(text, 0, 23)
 	if err != nil {
-		return 0, fmt.Errorf("unexpected hour:minute format: %w", err)
+		return 0, fmt.Errorf("could not parse hour: %s", err)
 	}
-	m, err := strconv.ParseInt(minute, 10, 64)
+	if len(text) <= hProgress || text[hProgress] != ':' {
+		return 0, fmt.Errorf("could not parse hour:minute format: character after hour [%s] is not a [:]", string(text))
+	}
+	hProgress += 1
+	mProgress, m, err := parseDigitRespectingOptionalPlaces(text[hProgress:], 0, 59)
```

Comment:
Same here

Reply:
Strings can be indexed at their ending boundary. The zero-length string will cause `parseDigitRespectingOptionalPlaces` to return an error.

```
	fmt.Println("123"[3:]) # empty string
	fmt.Println("123"[4:]) # index 4 out of bounds
```

Please see test case `parse_time with %R without minute element` for reproduction

Reply:
Strings can be indexed at their ending boundary. The zero-length string will cause `parseDigitRespectingOptionalPlaces` to return an error.

```
	fmt.Println("123"[3:]) # empty string
	fmt.Println("123"[4:]) # index 4 out of bounds
```

Please see test case `parse date with %F separator but no month` for reproduction

Reply:
OK, It seems to be ok for slice processing.